src/ansys/dpf/composites/failure_criteria/_failure_mode_enum.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+
+# Use the functional API because otherwise it is impossible to use
+# python keywords as enum values (such as "as")
+# Note: This enum needs to be in sync with the corresponding enum in dpf_composites
+FailureModeEnum = Enum(
```

Comment:
I'm guessing new failure modes could still be used, but would need to be explicitly referred to by their index?

For compatibility between versions, I guess this means we must never change existing indices (except maybe removing them), or re-use indices that were previously in use.